### PR TITLE
(WIP) a better version of enhancement and separation downstream

### PR DESCRIPTION
**Pull request by HuangZiliAndy**
Hi @leo19941227 , I am making the pull request for a better version of enhancement and separation downstream. In this pull request, I

- Add two new configs which have a much smaller model size and better performance
- Made some small changes to the code, including (1) modifying the loss function, supporting L1 loss, and computing loss in log domain (for smaller input scale and more stable training) (2) removing the original postprocess function. Originally, I found there are some issues when I am using librosa.istft, and I am using the postprocess function to remove the impulse at the end of the signal.  Now, I have found a better way to deal with this issue.

### CODE REVIEW

**Comment by HuangZiliAndy** [comment]
enh
sep
These two figures are comparisons of previous and new config. The top figure is for pesq value for enhancement task, and the bottom figure is for si-sdri value for the separation task. For enhancement, I think the previous config is not good due to mask type and activation function. The updated version is much better.

**Comment by HuangZiliAndy** [comment]
Detail changes for enhancement:

- remove total training steps from 150000 to 100000 (model generally converges with 100000 iterations)
- adjust the learning rate from 1e-4 to 1e-3 (compare different learning rates and 1e-3 is slightly better)
- adjust the win_length of STFT from 512 to 400 (a small change doesn't change the result, 400 is more usually used for ASR 25ms)
- reduce the model size from 896 to 256 (I found the influence of model size on pesq is extremely small)
- reduce the dropout rate from 0.5 to 0.1 (a small change that has little improvement)
- change the loss type from L2 to L1 (a small change that has little improvement)
- change the mask activation from relu to sigmoid and change the mask type from PSM to AM (the combination of mask type and activation function is the main reason for the improvement)

**Comment by HuangZiliAndy** [comment]
Detail changes for separation:

- adjust the learning rate from 1e-4 to 1e-3 (compare different learning rates and 1e-3 is better ~0.5dB improvement)
- **adjust n_fft and window_size of STFT from 512 to 1024 (for separation task, it seems a larger window size gives better performance)**
- reduce the model size from 896 to 256 (I haven't done experiments related to model size but I think a larger model size will have better performance, but following the principle of SUPERB, we should use a smaller model size)
- reduce the dropout rate from 0.5 to 0.1 (a small change that has little improvement)
- **change the mask activation from relu to sigmoid (~0.6dB improvement)**

**Comment by leo19941227** [comment]
Hey @HuangZiliAndy !

Thanks for the great PR! The performance improvement is huge!

I briefly look into it and I found there are quite some logic changes might be hard to make sure the backward compatibility. Do you think it is possible for you to create two new folders for the new better setting? So that we can always have the exactly same old setting, and the users won't be confused. We can use the folder names like `enhancement_stft2` or else.

What do you think?

**Comment by HuangZiliAndy** [comment]
Hi @leo19941227 , I agree with that. I think we need to add to README that enhancement_stft/enhancement_stft2 are working on the same datasets but with different configurations (also some tiny implementation differences).

**Comment by leo19941227** [comment]
Hey @HuangZiliAndy,

I can successfully run it. However, I find that you did not include any README in this PR. Would you like to merge it first or add the README before merging this PR?

**Comment by HuangZiliAndy** [comment]
Hi Leo, I will add a README this weekend before merging. Thanks!

**Comment by leo19941227** [comment]
Cool, thanks!

**Comment by leo19941227** [comment]
Hey @HuangZiliAndy !

You don't need to spend time solving this CI issue. All the PRs are now experiencing this issue and I am fixing this. You are facing this CI error since you merge the latest main so you are now using the latest github action. Anyway, I believe this PR can be smoothly merged after I fix the CI error.

**Comment by HuangZiliAndy** [comment]
Hi @leo19941227 , I have added the README. Could you please check if there is anything missing? Thanks!

**Comment by leo19941227** [comment]
LGTM. Thanks @HuangZiliAndy , I am merging this.
